app/Http/Controllers/Controller.php

```
@@ -98,23 +99,39 @@ public function link()
+            $email = Str::lower($email);
             $user = User::whereEmail($email)->first();
             if (! $user) {
                 return redirect()->route('login');
             }
+
+            $invitation = TeamInvitation::whereEmail($email)->first();
+            if (! $invitation || ! $invitation->isValid()) {
+                return redirect()->route('login')->with('error', 'Invitation has expired or been revoked.');
+            }
+
             if (Hash::check($password, $user->password)) {
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\Illuminate\Support\Facades\Hash' in method 'link'. (undefined)

(StaticAccess)